src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java

```diff
@@ -307,26 +307,29 @@ public void update() {
 				Matcher activeEffectsMatcher = PATTERN_ACTIVE_EFFECTS.matcher(line);
 				if (activeEffectsMatcher.matches()) {
 					foundGodPotText = true;
-					String[] godpotRemaingTimeUnformatted = activeEffectsMatcher.group(1).split(":");
 					long godPotDuration = 0;
 					try {
-						int i = 0;
-						if (godpotRemaingTimeUnformatted.length == 4) {
-							godPotDuration =
-								godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 24 * 60 * 60 * 1000;
-							i++;
-						}
-						if (godpotRemaingTimeUnformatted.length >= 3) {
-							godPotDuration =
-								godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 60 * 60 * 1000;
-							i++;
-						}
-						if (godpotRemaingTimeUnformatted.length >= 2) {
-							godPotDuration = godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 60 * 1000;
-							i++;
-						}
-						if (godpotRemaingTimeUnformatted.length >= 1) {
-							godPotDuration = godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 1000;
+						long godpotRemainingTime = Integer.parseInt(activeEffectsMatcher.group(1));
+						String godpotRemainingTimeType = activeEffectsMatcher.group(2);
+						switch (godpotRemainingTimeType) {
+							case "Years":
```

Comment:
P sure these could be the non plural versions like the cookie "Hour" etc

Reply:
implemented it like you in https://github.com/NotEnoughUpdates/NotEnoughUpdates/pull/432/commits/99a0fe53a9d2f0973dd210d5f022642c992d9ad5